bundles/org.openhab.binding.feed/src/main/java/org/openhab/binding/feed/internal/handler/FeedHandler.java

```diff
@@ -185,10 +238,43 @@ private void publishChannelIfLinked(ChannelUID channelUID) {
                 logger.debug("Unrecognized channel: {}", channelID);
         }
 
+        setChannelToState(channelUID, state);
+    }
+
+    private void setChannelToState(ChannelUID channel, State state) {
         if (state != null) {
-            updateState(channelID, state);
+            updateState(channel.getId(), state);
```

Comment:
The method called here `updateState` with channel id passed as String constructs a `ChannelUID` and than calls the method variant with `ChannelUID` parameter. So if you have already a `ChannelUID` it makes more sense to pass that directly:
```suggestion
            updateState(channel, state);
```